### DISCUSSION TITLE
Show combined TOC on overview page when navigation indexes is enabled

### DISCUSSION THREAD

**Discussion by SoumayaMauthoorMOJ** [squidfunk/mkdocs-material#5805] [Q&A] [answered]
When I enable [navigation indexes](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#:~:text=Navigation%20integration-,%C2%B6,-Permanent%20link) and split the TOC, the TOC for the overview page is still available:

```
  features:
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.indexes
```

![image](https://github.com/squidfunk/mkdocs-material/assets/95432492/50b63c02-a5bf-47bf-b79f-9eabff6ca43e)

However when I enable navigation indexes and merge the TOC with the navigation sidebar I lose the TOC from the overview page:

```
  features:
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.indexes 
    - toc.integrate
```

![image](https://github.com/squidfunk/mkdocs-material/assets/95432492/5cbd03ee-413c-4307-a5ae-72bea3e2ef6a)

Is there a way of keeping the TOC from the overview page even when merged with navigation sidebar?

**Comment by SoumayaMauthoorMOJ** [comment]
Perhaps I should clarify this only affects the overview page:

![image](https://github.com/squidfunk/mkdocs-material/assets/95432492/e3034e0e-e404-4cf8-9a9a-5b4de4207153)

**Comment by squidfunk** [comment, marked as answer]
Jup, [those features are currently incompatible](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages). It's unclear how we should make them work together 😅

<img width="736" alt="Bildschirmfoto 2023-08-05 um 17 41 00" src="https://github.com/squidfunk/mkdocs-material/assets/932156/f275aec8-4bf8-43d1-af68-a4a3847e2907">

**Reply by SoumayaMauthoorMOJ** [reply]
Oops I missed the callout :-)

It's a pity they're incompatible but thanks for clarifying!

**Reply by ssbarnea** [reply]
@squidfunk I know that you documented that `toc.integrate` and `navigation.indexes` are incompatible (on only one of them and in a hard to spot way due to the need to click on the bullet), but at the same time I see it possible to enable both without getting any visible errors.

If those features were by design not compatible, we should see a build error, just to avoid accidentally enabling them.

Based on my tests, it seems to be partially working with 9.5.2 version. By partially, I mean that the the index pages cannot have any toc sections displayed. Still, other child pages can have them, only the indexes are affected.

To my this means that one needs to be careful when using these two features combined to keep the indexes brief/small and put the beefy content in child pages. If user needs to scroll when inside the index page, probably you failed to made it small enough.

Am I right about this or I missed some essential side effects?

**Reply by squidfunk** [reply]
> If those features were by design not compatible, we should see a build error, just to avoid accidentally enabling them.

Those are entirely template features – we can't trigger build errors from templates. It works partially, yes, but it's undefined behavior from our side, so it's currently not supported. It might change in the future, but there are currently other priorities.
